pkg/interactor/save_data.go

```diff
@@ -17,7 +16,7 @@ func (d *DataInteractor) Save(token, id string, data []entities.Payload, timesta
 	for _, dt := range data {
 		err = d.DataStore.Save(entities.Data{From: id, Payload: dt, Timestamp: timestamp})
 		if err != nil {
-			return fmt.Errorf("error saving data %v: %w", data, err)
+			return ErrSaveDataDB
```

Comment:
I think that is not necessary, you should continue wraping the `DataStore.Save` error

Reply:
Done!